pandas/core/generic.py

```diff
@@ -9244,7 +9244,7 @@ def _tz_convert(ax, tz):
                 raise ValueError(f"The level {level} is not valid")
             ax = _tz_convert(ax, tz)
 
-        result = self._constructor(self._data, copy=copy)
+        result = self.copy(deep=copy)
```

Comment:
Do we need to at minimum always shallow copy here?

Reply:
isnt that what this is doing?

Reply:
Ah right! Fair.